src/main/java/org/dependencytrack/resources/v1/ProjectResource.java

```diff
@@ -873,8 +875,10 @@ public Response deleteProject(
                 }
 
                 LOGGER.info("Project " + project + " deletion request by " + super.getPrincipal().getName());
-                try {
-                    qm.recursivelyDelete(project, true);
+
+                try (final Handle jdbiHandle = openJdbiHandle()) {
+                    final var projectDao = jdbiHandle.attach(ProjectDao.class);
+                    projectDao.deleteProject(project.getUuid());
```

Comment:
In cases like this where there's already a transaction ongoing (see `qm.runInTransaction` above), we must avoid opening another one. Problem being that each transaction allocates a database connection until it commits or is rolled back. Holding two connections at once is too wasteful.

Two options:

* Execute the deletion query using the `QueryManager` in scope (`qm`) instead.
* Create a JDBI instance using [`JdbiFactory#createLocalJdbi`](https://github.com/DependencyTrack/hyades-apiserver/blob/425cabc125664d75c856f26ee0b6ab8ea19ffda6/src/main/java/org/dependencytrack/persistence/jdbi/JdbiFactory.java#L138-L158) (you'd need to make that method `public` first).